### PR TITLE
bug(ssim/ms-ssim): MS-SSIM feature was fixed to provide valid values.

### LINKED ISSUE #39: MS-SSIM returns wrong values
**Describe the bug**
`multi_scale_ssim` and `MultiScaleLoss` appear to estimate wrong values compared to results obtained from official TensorFlow implementation. 

**To Reproduce**
```python
from photosynthesis_metrics import multi_scale_ssim, MultiScaleSSIMLoss
import torch
import tensorflow as tf
from time import time
torch.manual_seed(42)
for _ in range(2):
    a = torch.rand(3, 3, 256, 256)
    b = torch.rand(3, 3, 256, 256)
    start = time()
    ps_ms_ssim = multi_scale_ssim(a,b, kernel_size=11, kernel_sigma=1.5, size_average=False, data_range=1.).numpy()
    ps_time = time()-start
    start = time()
    ps_ms_ssim_loss = MultiScaleSSIMLoss( kernel_size=11, kernel_sigma=1.5, data_range=1.)(a, b).numpy()
    ps_loss_time = time() - start
    tf_a = tf.convert_to_tensor(a.permute(0, 2, 3, 1).numpy())
    tf_b = tf.convert_to_tensor(b.permute(0, 2, 3, 1).numpy())
    start = time()
    tf_ms_ssim = tf.image.ssim_multiscale(tf_a, tf_b, max_val=1., filter_size=11, filter_sigma=1.5).numpy()
    tf_time = time() - start
    print(f'photosynthesis result: {ps_ms_ssim}, time: {ps_time}s')
    print(f'tensorflow result: {tf_ms_ssim}, time: {tf_time}s\n')
    print(f'photosynthesis loss result: {ps_ms_ssim_loss}, time: {ps_loss_time}s')
    print(f'tensorflow result: {tf_ms_ssim.mean()}, time: {tf_time}s\n')
```

**Output**
```python
>>>photosynthesis result: [0.06354865 0.06629971 0.07052894], time: 0.20274901390075684s
>>>tensorflow result: [0.05706326 0.08324481 0.08343431], time: 1.2265629768371582s

>>>photosynthesis loss result: 0.06679243594408035, time: 0.18335914611816406s
>>>tensorflow result: 0.07458078861236572, time: 1.2265629768371582s

>>>photosynthesis result: [0.06800205 0.07029662 0.07833724], time: 0.19111299514770508s
>>>tensorflow result: [0.08063222 0.08336634 0.09324924], time: 1.2148618698120117s

>>>photosynthesis loss result: 0.07221197336912155, time: 0.20091605186462402s
>>>tensorflow result: 0.08574926853179932, time: 1.2148618698120117s
```

**Expected behavior**
Same values are expected.

**Additional context**
The bug was found during inspection of issue #13 . I assume the problem is located at `_compute_multi_scale_ssim`. Possible fail points:
1. In the TensorFlow implementation of MS-SSIM, [SSIM](https://github.com/photosynthesis-team/photosynthesis.metrics/blob/e17c4c1ddc19f282b12b232e4332f27b3291ea49/photosynthesis_metrics/ssim.py#L458) appears to return tensor of (batch, channels), while in current implementation tensor has only (batch) dimension.
2.  First, [the product](https://github.com/photosynthesis-team/photosynthesis.metrics/blob/e17c4c1ddc19f282b12b232e4332f27b3291ea49/photosynthesis_metrics/ssim.py#L475) should be estimated. Then, the result should be averaged along channel dimension. Current implementation performs it vice versa.

### PR DESCRIPTION
**Pull request by denproc**
Closes #39 and #13.

### CODE REVIEW

**Comment by denproc** [comment]
Rebased onto current master. I suggest to merge this PR first, then #38.

File: photosynthesis_metrics/ssim.py [hunk 1]
```diff
@@ -394,6 +429,49 @@ def _fspecial_gauss_1d(size: int, sigma: float) -> torch.Tensor:
 
     return g.unsqueeze(0).unsqueeze(0)
```
Comment (snk4tr) [inline photosynthesis_metrics/ssim.py:431]: Formatting: two lines between top-level functions/classes (see [Google style guide](http://google.github.io/styleguide/pyguide.html))

File: photosynthesis_metrics/ssim.py [hunk 2]
```diff
@@ -394,6 +429,49 @@ def _fspecial_gauss_1d(size: int, sigma: float) -> torch.Tensor:
...
+    sigma12 = compensation * (_gaussian_filter(x * y, kernel) - mu1_mu2)
+
+    # Set alpha = beta = gamma = 1.
+    cs_map = (2 * sigma12 + c2) / (sigma1_sq + sigma2_sq + c2)
+    ssim_map = ((2 * mu1_mu2 + c1) / (mu1_sq + mu2_sq + c1)) * cs_map
+
+    ssim_val = ssim_map.mean(dim=(-1, -2))
+    cs = cs_map.mean(dim=(-1, -2))
+
+    return ssim_val, cs
```
Comment (snk4tr) [inline photosynthesis_metrics/ssim.py:475]: Formatting: two lines between top-level functions/classes (see [Google style guide](http://google.github.io/styleguide/pyguide.html))

File: photosynthesis_metrics/utils.py [hunk 3]
```diff
@@ -31,10 +30,9 @@ def _validate_input(x: torch.Tensor, y: torch.Tensor, kernel_size: Optional[int]
     if kernel_size is not None:
         assert kernel_size % 2 == 1, f'Kernel size must be odd, got {kernel_size}.'
     if scale_weights is not None:
-        assert isinstance(scale_weights, (list, tuple)), \
+        assert isinstance(scale_weights, (list, tuple, torch.Tensor)), \
             f'Scale weights must be of type list or tuple, got {type(scale_weights)}.'
```
Comment (snk4tr) [inline photosynthesis_metrics/utils.py:34]:
```suggestion
        assert isinstance(scale_weights, (list, tuple, torch.Tensor)), \
            f'Scale weights must be of type list, tuple or torch.Tensor, got {type(scale_weights)}.'
```

**Comment by denproc** [comment]
As new tests were introduced to compare our implementation with official TensorFlow, the following bug was detected. The previous version of MS-SSIM appears to estimate wrong values due to incorrect evaluation of SSIM at each scale and final aggregation of the results. More details are available at #39.
This goals of this PR are to fix the MS-SSIM functionality and to provide test scenarios.

**Pull request merged**